src/transform/codegen/ode_factory.jl

```diff
@@ -0,0 +1,144 @@
+    @ccall jl_safe_printf("$key\n"::Cstring)::Cvoid
+    torn_ci = find_matching_ci(ci->isa(ci.owner, TornIRSpec) && ci.owner.key == key, ci.def, world)
+    torn_ir = torn_ci.inferred
+
+    (;ir_sicm) = torn_ir
+
+    ir_factory = copy(result.ir)
+    pushfirst!(ir_factory.argtypes, Settings)
+    pushfirst!(ir_factory.argtypes, typeof(factory))
+    compact = IncrementalCompact(ir_factory)
+
+    local line
+    if ir_sicm !== nothing
+        sicm_ci = find_matching_ci(ci->isa(ci.owner, SICMSpec) && ci.owner.key == key, ci.def, world)
+        @assert sicm_ci !== nothing
+
+        line = result.ir[SSAValue(1)][:line]
+        #insert_node_here!(compact, NewInstruction(Expr(:call, println, "Trace: A"), Cvoid, line))
+        sicm = insert_node_here!(compact,
+            NewInstruction(Expr(:call, invoke, Argument(3), sicm_ci, (Argument(i+1) for i = 2:length(result.ir.argtypes))...), Tuple, line))
+        #insert_node_here!(compact, NewInstruction(Expr(:call, println, "Trace: B"), Cvoid, line))
+    else
+        sicm = ()
+    end
+
+    odef_ci = rhs_finish!(result, ci, key, world, 1)
+
+    # Create a small opaque closure to adapt from SciML ABI to our own internal
+    # ABI
+
+    numstates = zeros(Int, Int(LastEquationStateKind))
+
+    all_states = Int[]
+    for var = 1:length(result.var_to_diff)
+        kind = classify_var(result.var_to_diff, key, var)
+        kind == nothing && continue
+        @ccall jl_safe_printf("$kind\n"::Cstring)::Cvoid
+        numstates[kind] += 1
+        (kind != AlgebraicDerivative) && push!(all_states, var)
+    end
+
+    ir_oc = copy(result.ir)
+    empty!(ir_oc.argtypes)
+    argt = Tuple{Vector{Float64}, Vector{Float64}, SciMLBase.NullParameters, Float64}
+    push!(ir_oc.argtypes, Tuple)
+    append!(ir_oc.argtypes, fieldtypes(argt))
+
+    oc_compact = IncrementalCompact(ir_oc)
+    self = Argument(1)
+    du = Argument(2)
+    u = Argument(3)
+    p = Argument(4)
+    t = Argument(5)
+
+    # Zero the output
+    line = ir_oc[SSAValue(1)][:line]
+
+    @insert_node_here oc_compact line zero!(du)::VectorViewType
+
+    # out_du_mm, out_eq, in_u_mm, in_u_unassgn, in_alg
+    nassgn = numstates[AssignedDiff]
+    ntotalstates = numstates[AssignedDiff] + numstates[UnassignedDiff] + numstates[Algebraic]
+    out_du_mm = @insert_node_here oc_compact line view(du, 1:nassgn)::VectorViewType
+    out_eq = @insert_node_here oc_compact line view(du, (nassgn+1):ntotalstates)::VectorViewType
+
+    (in_u_mm, in_u_unassgn, in_alg) = sciml_ode_split_u!(oc_compact, line, u, numstates)
+
+    # Call DAECompiler-generated RHS with internal ABI
+    oc_sicm = @insert_node_here oc_compact line getfield(self, 1)::Tuple
+
+    # N.B: The ordering of arguments should match the ordering in the StateKind enum
+    @insert_node_here oc_compact line (:invoke)(odef_ci, oc_sicm, (), in_u_mm, in_u_unassgn, in_alg, out_du_mm, out_eq, t)::Nothing
```

Comment:
This is missing an argument for the DAECompiler ABI

Reply:
`in_du_unassgn` to be set to `nothing`?

Reply:
No, `u` should have 4 partitions

Reply:
Ah, got it

Reply:
Seems like it fixed the segfault, thanks a lot for pointing that out.